daemon/daemon.go

```diff
@@ -127,7 +127,7 @@ func (d *Daemon) UpdateProxyRedirect(e *endpoint.Endpoint, l4 *policy.L4Filter)
 		return 0, fmt.Errorf("can't redirect, proxy disabled")
 	}
 
-	r, err := d.l7Proxy.CreateOrUpdateRedirect(l4, e.ProxyID(l4), e, d)
+	r, err := d.l7Proxy.CreateOrUpdateRedirect(l4, e.ProxyID(l4), e, d, e)
```

Comment:
Why do we need to pass `e` twice, as well as `e.ProxyID(l4)` separately?

Reply:
'e', endpoint is being passed in different roles (literally different interfaces), the called function does not (need to) know that they are the same 'e':

```
func (p *Proxy) CreateOrUpdateRedirect(l4 *policy.L4Filter, id string, source ProxySource,
	notifier accesslog.LogRecordNotifier, completions policy.CompletionContainer) (Redirect, error)
```